tests/kafkatest/tests/core/network_degrade_test.py

```diff
@@ -0,0 +1,117 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from ducktape.mark import parametrize
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+
+from kafkatest.services.trogdor.degraded_network_fault_spec import DegradedNetworkFaultSpec
+from kafkatest.services.trogdor.trogdor import TrogdorService
+from kafkatest.services.zookeeper import ZookeeperService
+
+
+class NetworkDegradeTest(Test):
+    """
+    This is a sanity test to make sure tc is behaving like we expect in our test environment
+    """
+
+    def __init__(self, test_context):
+        super(NetworkDegradeTest, self).__init__(test_context)
+        self.zk = ZookeeperService(test_context, num_nodes=3)
+        self.trogdor = TrogdorService(context=self.test_context, client_services=[self.zk])
+
+    def setUp(self):
+        self.zk.start()
+        self.trogdor.start()
+
+    def teardown(self):
+        self.trogdor.stop()
+        self.zk.stop()
+
+    @parametrize(task_name="latency-100", latency_ms=100, rate_limit_kbit=0)
+    @parametrize(task_name="latency-100-rate-1000", latency_ms=100, rate_limit_kbit=1000)
+    def test_latency(self, task_name, latency_ms, rate_limit_kbit):
+        spec = DegradedNetworkFaultSpec(0, 10000, {})
+        for node in self.zk.nodes:
+            spec.add_node_spec(node.name, "eth0", latency_ms, rate_limit_kbit)
+
+        latency = self.trogdor.create_task(task_name, spec)
+
+        zk0 = self.zk.nodes[0]
+        zk1 = self.zk.nodes[1]
+
+        r = re.compile(r".*time=(?P<time>[\d.]+)\sms.*")
+        times = []
+        for line in zk0.account.ssh_capture("ping -i 1 -c 20 %s" % zk1.name):
+            self.logger.debug("Ping output: %s" % line)
+            m = r.match(line)
+            if m is not None and m.group("time"):
+                times.append(float(m.group("time")))
+                self.logger.debug("Parsed ping time of %d" % float(m.group("time")))
+
+        # We expect to see some low ping times (before and after the task runs)
+        # as well as high ping times (during the task)
+        slow_times = [t for t in times if t > 100]
+        fast_times = [t for t in times if t < 10]
```

Comment:
The problem with this one is that the `ping` output are from a single request. If the trogdor task has started, but the `tc` command hasn't finished, and the first `ping` fires -- it will return a high response time and fail an assertion. Rather than go this route, I opted to capture samples before during and after.

Alternatively, we could add a sleep after waiting for trogdor to be running. I did reduce the expected number of "slow" and "fast" times to decrease the sensitivity a bit.